src/tests/ftest/dfuse/pil4dfs_fio.py

```diff
@@ -0,0 +1,223 @@
+from general_utils import bytes_to_human
+
+
+class Pil4dfsFio(DfuseTestBase):
+    """Test class Description: Runs Fio with in small config.
+
+    :avocado: recursive
+    """
+
+    _FIO_RW_NAMES = ["write", "read"]
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a FioPil4dfs object."""
+        super().__init__(*args, **kwargs)
+
+        self.fio_cmd = None
+        self.fio_params = {"thread": "", "blocksize": "", "size": ""}
+        self.fio_numjobs = 0
+        self.fio_cpus_allowed = ""
+
+    def setUp(self):
+        """Set up each test case."""
+        # obtain separate logs
+        self.update_log_file_names()
+
+        # Start the servers and agents
+        super().setUp()
+
+        for name in self.fio_params:
+            self.fio_params[name] = self.params.get(name, "/run/fio/job/*", "")
+
+        cpu_info = CpuInfo(self.log, self.hostlist_clients)
+        cpu_info.scan()
+        _, arch = cpu_info.get_architectures()[0]
+        if arch.numas != 2:
+            self.fail(f"Client with unsupported quantity of NUMA nodes: wait=2, got={arch.numas}")
+        self.fio_numjobs = int(arch.quantity / arch.threads_core)
+        cpus = []
+        cores_quantity = int(self.fio_numjobs / 2)
+        for numa_idx in range(2):
+            cpus += arch.get_numa_cpus(numa_idx)[:cores_quantity]
+        self.fio_cpus_allowed = str(NodeSet(str(cpus)))[1:-1]
+
+    def create_container(self):
+        """Created a DAOS POSIX container"""
+        self.log.info("Creating pool")
+        self.assertIsNone(self.pool)
+        self.add_pool()
+        self.log.debug("Created pool %s", str(self.pool))
+
+        self.log.info("Creating container")
+        self.assertIsNone(self.container)
+        self.add_container(self.pool)
+        self.log.debug("Created container %s", str(self.container))
+
+    def destroy_container(self):
+        """Destroy DAOS POSIX container previously created"""
+        self.assertIsNotNone(self.container)
+        self.log.debug("Destroying container %s", str(self.container))
+        self.destroy_containers(self.container)
+        self.container = None
+
+        self.assertIsNotNone(self.pool)
+        self.log.debug("Destroying pool %s", str(self.pool))
+        self.destroy_pools(self.pool)
+        self.pool = None
+
+    def get_bandwidth(self, fio_result, rw):
+        """Returns FIO bandwidth of a given I/O pattern
+
+        Args:
+            fio_result (RemoteCommandResult): results of a FIO command.
+            rw (str): Type of I/O pattern.
+
+        Returns:
+            int: Bandwidth of the FIO command.
+
+        """
+        fio_stdout = next(iter(fio_result.all_stdout.values()))
+        # NOTE With dfuse and pil4dfs some junk messages could be eventually printed
+        if fio_stdout[0] != '{':
+            fio_stdout = '{' + fio_stdout.partition('{')[2]
+        fio_json = json.loads(fio_stdout)
+        return fio_json["jobs"][0][rw]['bw_bytes']
```

Comment:
There is only maybe junk before `{`, but not after `}`?

Reply:
Until now, the junk that I have observed was only at the beginning of the output.
Moreover, the junk should have been removed with the latest modification from @wiliamhuang.